### PR TITLE
Change HAL pin type for hy_gt_vfd.fault-info from float to u32

### LINKED ISSUE #2206: HY_GT_VFD fault info register
The HAL component HY_GT_VFD declares a HAL pin called hy_gt_vfd.fault-info as a FLOAT output. However, that register (0x5000) is a set of BIT values and should therefore be declared as a U32.  Then it will be possible for python to convert it to a hex value for display in a GUI.

### PR DESCRIPTION
**Pull request by petterreinholdtsen**
Restructure code to handle two types of HAL pin types for modbus register.

Fixes #2206.

### CODE REVIEW

**Comment by petterreinholdtsen** [comment]
I would like someone who know this component a little better to see if this is a wanted change and if the approach taken is a good one.

**Comment by andypugh** [comment]
Looks reasonable, though possibly the input_terminal should be int too. (but then I suspect that output_terminal isn't really handled at all in any useful way)

**Comment by SebKuzminsky** [comment]
Looks good to me.  I no longer have access to this VFD to test it, I'm interested to see if it works for @persei802.

I also don't have access to the manual, but I wonder if the fault-info register is a bitmap of different errors; if so it might make sense to break it out into a bunch of bit pins identifying the different errors.

**Comment by persei802** [comment]
That is correct. There are 26 bits indicating a fault condition. In my GUI, I convert this register to a hex value so it's easy to determine which bits are set. The manual doesn't actually explain what each of these bits mean, except that 0x00 is no fault. In my personal opinion, it is not useful to break them out into individual HAL pins.
Jim

File: docs/src/man/man1/hy_gt_vfd.1.adoc [hunk 1]
```diff
@@ -257,9 +257,11 @@ PC.03 = 1::
 
         The VFD's external counter register.
 
-*hy_gt_vfd.fault-info* (float, out)::
+*hy_gt_vfd.fault-info* (u32, out)::
 
-        The VFD's fault info register.
+        The VFD's fault info register.  The type of this register was
+        changed from float to u32 in version 2.10.  The commit was
+        done 2022-12-17.
```
Comment (SebKuzminsky) [inline docs/src/man/man1/hy_gt_vfd.1.adoc:264]: This kind of user-facing changelog information belongs in [Updating LinuxCNC](http://linuxcnc.org/docs/devel/html/getting-started/updating-linuxcnc.html), not in the manpage.

**Review by SebKuzminsky — CHANGES_REQUESTED** [review]
Looks good, except for the nitpick about changelog location.

**Comment by c-morley** [comment]
Could this be done with s32 rather then u32 or do you need to extra range? 
u32 is awkward to use in many components ie classicladder.

**Comment by andypugh** [comment]
I think U32 is the right choice if the intention is to break it out into bits. It matches the bitslice component. 
It's a bitmap, so signed does not make any sense (even if it is only 16 bits)

**Comment by c-morley** [comment]
Then maybe have both pins? HAL pins are cheap.

**Comment by petterreinholdtsen** [comment]
[Sebastian Kuzminsky]
> @SebKuzminsky requested changes on this pull request.
>
> Looks good, except for the nitpick about changelog location.
>
>> +        The VFD's fault info register.  The type of this register was
> +        changed from float to u32 in version 2.10.  The commit was
> +        done 2022-12-17.
>
> This kind of user-facing changelog information belongs in [Updating
> LinuxCNC](http://linuxcnc.org/docs/devel/html/getting-started/updating-linuxcnc.html),
> not in the manpage.

The idea is to make sure those using the fault-info value today who
experience unexpected behaviour when they upgrade to 2.10, should be
able to find a clue why in the man page of the failing module.

I find it valuable when information about when a feature was introduced
is included in manual pages, like it is for example in several linux
syscall man pages and debhelper(7).

I have nothing against listing it in
getting-started/updating-linuxcnc.adoc.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by petterreinholdtsen** [comment]
[c-morley]
> Then maybe have both pins? HAL pins are cheap.

I left the old fault-info float pin alone, and introduced a new
fault-info-bits pin instead, to stay backwards compatible while also
providing a u32 with easier interpretation of the error bits.  Updated
documentation to match.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by c-morley** [comment]
Ahh I see, that works too.

**Comment by persei802** [comment]
I misspoke earlier. This register has 26 fault codes, not bits. It is a 16 bit register. Anyway, I ran Petter's modified code and it works fine. I have no further issues with this component.

File: docs/src/man/man1/hy_gt_vfd.1.adoc (hunk 1 above)
Comment (SebKuzminsky) [reply docs/src/man/man1/hy_gt_vfd.1.adoc]: The patch was reworked to not break backward compatibility, so no note is needed in `Updating LinuxCNC`.

**Comment by petterreinholdtsen** [comment]
[Jim]
> I misspoke earlier. This register has 26 fault codes, not bits. It is
> a 16 bit register. Anyway, I ran Petter's modified code and it works
> fine. I have no further issues with this component.

Aha.  Adjusted documentation to drop the 'bit' claim.  Where is the list
of fault codes documnented?

-- 
Happy hacking
Petter Reinholdtsen

**Comment by persei802** [comment]
They are in the GT Series Inverter Manual.
fault_codes

**Comment by petterreinholdtsen** [comment]
[Jim]
> They are in the GT Series Inverter Manual.

Thank you.  Added this reference to the man page, and renamed the new
u32 pin from fault-info-bits to fault-info-code, as it is not a bit
field.

I believe this is ready for merging.
-- 
Happy hacking
Petter Reinholdtsen

**Pull request merged**